ExcelReaderAPI/Services/FileUploadService.cs

```diff
@@ -0,0 +1,231 @@
+﻿using System.Data;
+using System.Data.Common;
+using System.Data.SqlClient;
+using System.Globalization;
+using ExcelDataReader;
+using ExcelReaderAPI.Models;
+
+namespace ExcelReaderAPI.Services
+{
+
+
+    public class FileUploadService
+    {
+        private readonly DatabaseHelper _databaseHelper;
+
+        public FileUploadService(DatabaseHelper databaseHelper)
+        {
+            _databaseHelper = databaseHelper;
+        }
+
+        public void AddUserFiles(Guid fileId, string filename, SqlConnection connection)
+        {
+            const string reqAuthor = "Sikandar R";
+            DateTime uploadDate = DateTime.Now;
+
+            string userInsertionQuery =
+                "insert into ExcelReaderDb2.dbo.UserFiles values" +
+                "(@fileId, @fileName, @author, @uploadDate)";
+
+            var cmd = new SqlCommand(userInsertionQuery, connection);
+            cmd.Parameters.AddWithValue("@fileId", fileId);
+            cmd.Parameters.AddWithValue("@fileName", filename);
+            cmd.Parameters.AddWithValue("@author", reqAuthor);
+            cmd.Parameters.AddWithValue("@uploadDate", uploadDate);
+            try
+            {
+                cmd.ExecuteNonQuery();
+            }
+            catch
+            {
+                throw;
+            }
+
+        }
+
+        public string UploadExcelFile(IFormFile file)
+        {
+            string uploadFileResponse;
+            string fileName = file.FileName;
+            List<ITRequest> ITRequestsFromFile;
```

Comment:
Move this near to initialization and use var

Reply:
dont understand this

Reply:
make line 75 something like this

var ItRequestsFromFile = ExtractDataFromExcelFile(file, sourceFileId);